full-service/src/service/transaction.rs

```diff
@@ -512,6 +519,11 @@ where
             if let Some(inputs) = input_txo_ids {
                 builder.set_txos(conn, inputs)?;
             } else {
+                if let Some(subaddress) = spend_only_from_subaddress {
+                    // Ensure the builder will filter to txos only from the specified subaddress
+                    builder.set_spend_only_from_subaddress(subaddress)?;
```

Comment:
We set this on a property of the builder so that when the builder is filtering txos, it knows to only select from txos received at the given subaddress